src/Compilers/CSharp/Portable/FlowAnalysis/NullableWalker.SnapshotManager.cs

```diff
@@ -78,20 +74,21 @@ internal void VerifyNode(BoundNode node)
                     return;
                 }
 
-                Debug.Assert(_incrementalSnapshots.ContainsKey(node.Syntax.SpanStart), $"Did not find a snapshot for {node} `{node.Syntax}.`");
-                Debug.Assert(_walkerGlobalStates.Length > _incrementalSnapshots[node.Syntax.SpanStart].SharedStateIndex, $"Did not find global state for {node} `{node.Syntax}`.");
-            }
-#endif
+                int nodePosition = node.Syntax.SpanStart;
+                bool foundSnapshot = false;
+                foreach (var (position, snapshot) in _incrementalSnapshots)
```

Comment:
>foreach (var (position, snapshot) in _incrementalSnapshots) [](start = 16, length = 59)

Can we use binary search here too?

Reply:
Yeah, I'll change that.

---
In reply to: [295506083](https://github.com/dotnet/roslyn/pull/36563#discussion_r295506083) [](ancestors = 295506083)